docs/hugo/content/contributing/add-a-new-code-generated-resource/create-a-sample.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Create a sample to demonstrate the resource
+linktitle: Create a sample
+weight: 50
+---
+
+We mandate that every resource has a sample that demonstrates how to use it - and enforce this with checks in our CI pipeline. Good examples help other ASO users understand how to use each resource. 
+
+To ensure that each sample is correct, we run them - allowing us to catch any typos or errors before they trip up users in the wild.
+
+The samples are located in the [samples directory](https://github.com/Azure/azure-service-operator/blob/main/v2/samples). There should be at least one sample for each kind of supported resource. These currently need to be added manually. It's possible in the future we will automatically generate samples similar to how we automatically generate CRDs and types, but that doesn't happen today.
+
+The added new sample needs to be tested and recorded.
+
+If a recording for the test already exists, delete it. Look in the [recordings directory](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/recordings/Test_Samples_CreationAndDeletion) for a file with the same name as your new test.  
```

Comment:
How would this exist if it's a new sample? Or is the same test as in the previous step used to test the sample?

Reply:
Good point of clarification. There's a single go test for each subdirectory in the samples dir, so all resources for a given API version of say, network, are tested in one recording file.

This is because a lot of resources have dependencies between them, and trying to have each _resource_ have its own recording would result in a combinatoric set of tests/recordings.